storage/dataflux/range_splitter.go

```diff
@@ -216,6 +216,7 @@ func (rs *rangeSplitter) addCharsToAlphabet(characters []rune) {
 		if _, exists := rs.alphabetMap[char]; !exists {
 			allAlphabet = append(allAlphabet, char)
 			newChars = true
+			rs.alphabetMap[char] = 0
```

Comment:
I thought we previously removed this line, was it accidentally added back in?

Reply:
I ran a few tests and without this line, range_splitter was returning unexpected results.

Reply:
What tests did you run? I ran the ones in `range_splitter_test` in this PR and they pass fine without this line.

Reply:
So if I remove this line, and `characters` contains same new alphabet multiple times,  then it will be appended to  `allAlphabet` multiple times as the check to append is `if exists := rs.alphabetMap[char]; !exists{}`. So we will have repeated characters in allAlphabet and alphabetMap will not be as expected. 

Added a unit to cover this.

Reply:
This feels like the wrong place to check for this.
If it's an error for the same character to be added several times then we should return an error.

At the very least add a comment to this line please to indicate that it's there to avoid repeating the same character more than once.

Please fix the comment of the method too `// addCharsToAlphabet adds the given chars to the known alphabet. Repeated chars are ignored [or it's an error to repeat chars]`

Reply:
The `alphabet` has only unique characters (similar to english alphabets have unique characters) and the main goal of this method is to add new characters to `alphabet`. 
Updated newRangeSplitter comment to show that alphabets are unique chars. 
Update comments to explain the reason behind adding the char to the map.